test/end-to-end/company.js

```diff
@@ -1933,4 +1933,294 @@ describe("Company endpoint", () => {
+                        bio: "Without login",
+                        contacts: ["1"],
+                        logo: "http://awebsite.com/otherlogo.jpg",
+                    })
+                    .expect(HTTPStatus.UNAUTHORIZED);
+
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.INSUFFICIENT_PERMISSIONS);
+            });
+        });
+
+        describe("Should pass if good user", () => {
+            test("Should pass if god", async () => {
+                await test_agent
+                    .put(`/company/${test_company._id}/edit`)
+                    .send(withGodToken({
+                        name: "Changing Company",
+                        bio: "As god",
+                        contacts: ["1"],
+                        logo: "http://awebsite.com/otherlogo.jpg",
+                    }))
+                    .expect(HTTPStatus.OK);
+            });
+
+            test("Should pass if same company", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_company)
+                    .expect(HTTPStatus.OK);
+
+                const res = await test_agent
+                    .put(`/company/${test_company._id}/edit`)
+                    .send({
+                        name: "Changing Company",
+                        bio: "As company itself",
+                        logo: "http://awebsite.com/otherlogo.jpg",
+                    })
+                    .expect(HTTPStatus.OK);
+
+                expect(res.body).toHaveProperty("name", "Changing Company");
+                expect(res.body).toHaveProperty("bio", "As company itself");
+                expect(res.body).toHaveProperty("logo", "http://awebsite.com/otherlogo.jpg");
+            });
+        });
+
+        test("Offer should be updated", async () => {
+            await test_agent
+                .post("/auth/login")
+                .send(test_user_company)
+                .expect(HTTPStatus.OK);
+
+            const res = await test_agent
+                .put(`/company/${test_company._id}/edit`)
+                .send({
+                    name: "Changing Company",
+                    contacts: ["122"],
+                })
+                .expect(HTTPStatus.OK);
+
+            test_offer = await Offer.findById(test_offer._id);
+            expect(res.body).toHaveProperty("contacts", ["122"]);
+            expect(test_offer.ownerName).toEqual("Changing Company");
+            expect(test_offer.contacts).toEqual(["122"]);
+        });
+
+        describe("Should fail if company is blocked or disabled", () => {
+            test("Should fail if company is blocked (admin)", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_admin)
+                    .expect(HTTPStatus.OK);
+
+                await Company.findByIdAndUpdate(test_company._id, { isBlocked: true });
```

Comment:
This is fine but it would be best, or at least consistent with other tests, if you created a separate company that starts out as blocked.